hrana/test/hrana_test.dart

```diff
@@ -10,107 +10,113 @@ void main() {
-      final result = await stored.select();
-      expect(result.rows, [
-        [1]
-      ]);
-    });
+      tearDownAll(() async {
+        await server.stop();
+      });
 
-    test("don't work after being closed", () async {
-      final stored = await database.storeSql('SELECT 1');
-      await stored.close();
-      expect(() => stored.execute(), throwsStateError);
-    });
-  });
-
-  group('batch', () {
-    test('can run statements', () async {
-      final storedSelect = await database.storeSql('SELECT * FROM users;');
-      final storedExecute =
-          await database.storeSql('INSERT INTO users (name) VALUES (?);');
-
-      final results = await database.batch((builder) {
-        builder
-          ..executeStored(storedExecute, arguments: ['first'])
-          ..select('SELECT * FROM users;')
-          ..execute('INSERT INTO users (name) VALUES (?)',
-              arguments: ['second'])
-          ..selectStored(storedSelect);
+      setUp(() async {
+        database = await Database.connect(
+          Uri.parse('$scheme://localhost:$port/'),
+        );
+
+        await database.execute(
+            'CREATE TABLE users (name TEXT, CHECK (LENGTH(name) < 10));');
       });
 
-      final firstSelect = results.selectResult(1 as BatchRequest);
-      final secondSelect = results.selectResult(3 as BatchRequest);
+      tearDown(() async {
+        await database.execute('DROP TABLE users;');
 
-      expect(firstSelect!.rows, [
-        ['first']
-      ]);
-      expect(secondSelect!.rows, [
-        ['first'],
-        ['second']
-      ]);
-    });
+        await database.close();
+      });
+
+      test('can select statements', () async {
+        final result = await database.select('SELECT 1 AS r;');
+        expect(result.columnNames, ['r']);
+        expect(result.rows, [
+          [1]
+        ]);
+      });
 
-    test('does not run statements after failure', () async {
-      final results = await database.batch((builder) {
-        builder
-          ..execute('INSERT INTO users (name) VALUES (?);', arguments: [
-            'very long name that will be rejected by the check constraint'
-          ])
-          ..execute('SELECT * FROM users;');
+      test('can execute statements', () async {
+        final result = await database.execute(
+          'INSERT INTO users (name) VALUES (?);',
+          arguments: ['name'],
+        );
+        expect(result.affectedRows, 1);
+        expect(result.lastInsertRowId, 1);
       });
 
-      expect(() => results.executeResult(0 as BatchRequest),
-          throwsA(isA<ServerException>()));
-      expect(results.selectResult(1 as BatchRequest), isNull);
+      group('stored statements', () {
+        test('can be executed', () async {
+          final stored =
+              await database.storeSql('INSERT INTO users (name) VALUES (?);');
+          final result = await stored.execute(arguments: ['Stored']);
+          expect(result.affectedRows, 1);
```

Comment:
If the last insert rowid thing has been fixed, can we add an `expect(result.lastInsertRowId, 1)` here as well?

Reply:
Done 👍